lib/src/configuration.dart

```diff
@@ -147,6 +147,18 @@ abstract class Configuration {
         syncErrorHandler: syncErrorHandler,
         syncClientResetErrorHandler: syncClientResetErrorHandler,
       );
+
+  /// Constructs a [SessionlessSyncConfiguration]
+  static SessionlessSyncConfiguration sessionlessSync(
```

Comment:
Do we need another ctor for this. Why not expose a flag on the `local` ctor instead. This is really is useful in other scenarios as well. I believe this type of open should be exposed on the local ctor so we support all local open realm settings. 
I will add a discussion point about this since it seems we need to discuss other items related to this as well.

Reply:
I already had a discussion with @nirinchev about this. Since we are opening a realm with history, it is really closer in nature to a flexible sync realm than a local realm. There are a number of features that are not exposed on synchronizable realms, such as 

- `initialDataCallback`,
- `schemaVersion`,
- `disableFormatUpgrade`, and
- `shouldCompactCallback`

That would be odd, if not dangerous to expose, if the realm is opened simultaneously in another process with a flexible  sync configuration. It might make sense to expose `isReadOnly`, though.

Reply:
we settled on `disconnectedSync` constructor and type